scripts/serve_header/README.md

```diff
@@ -0,0 +1,91 @@
+serve_header.py
+===============
+
+Serves the `single_include/nlohmann/json.hpp` header file over HTTP(S).
+
+The header file is automatically amalgamated on demand.
+
+![serve_header.py demo](https://user-images.githubusercontent.com/320854/165481390-d65a5180-00d3-443b-b2f1-ea227316309f.png)
+
+## Prerequisites
+
+1. Make sure these Python packages are installed.
+    ```
+    PyYAML
+    watchdog
+    ```
+    (see `scripts/serve_header/requirements.txt`)
+
+2. To serve the header over HTTPS (which is required by Compiler Explorer at this time), a certificate is needed.
+   The recommended method for creating a locally-trusted certificate is to use [`mkcert`](https://github.com/FiloSottile/mkcert).
+   - Install the `mkcert` certificate authority into your trust store(s):
+     ```
+     $ mkcert -install
+     ```
+   - Create a certificate for `localhost`:
+     ```
+     $ mkcert localhost
+     ```
+     The command will create two files, `localhost.pem` and `localhost-key.pem`, in the current working directory. It is recommended to create them in the top level or project root directory.
```

Comment:
Well, I really want the configuration in the top level directory because I may want to read the configuration from other working trees at some point (for aliases; haven't decided yet what level of flexibility and complexity I want/need).
Also, I prefer these generated files to be (more) visible but the location is ultimately configurable by the user. We're just talking about the default and for simplicity I chose the project root for everything.

I also don't think the `scripts` folder belongs into `test` conceptually (who knows if future scripts will). If you want to unclutter the root, maybe we can trade and consider some other candidates:
* `benchmarks` into `test`
* `CODE_OF_CONDUCT.md` into `.github`; add a paragraph to the main `README.md` (similar to `CONTRIBUTING.md`)
* `CITATION.cff` into `doc`

I don't know if `nlohmann_json.natvis` or `wsjcpp.yml` could be moved.

Reply:
Moving benchmarks to tests is a good idea. For the other files, I think the must be in the root directory for GitHub to detect and link them automatically - I need to double check.

Moving the script to the tests folder was an idea given the title "enable rapid testing" :-)

Reply:
> Moving the script to the tests folder was an idea given the title "enable rapid testing" :-)

I figured. :-)
When I used "testing" in the title I meant it in the sense of "to try out". Different from "unit testing".

Reply:
You see, the test folder currently holds all kind of stuff: unit tests, fuzz tests, and cmake tests. It's a bit the catch-all folder for anything that is not production source code (include, single_include), documentation (doc), or other stuff (third_party). Maybe the last (third_party) could also be a good place if we rename it:

- amalgamate and cpplint are actually third party
- gdb_pretty_printer was built for the library and copied to the repo from an issue answer
- macro_builder is just some helpful source code

Reply:
How about either renaming `third_party` to `contrib` or moving it into `contrib` (and then disentangle what's truly third party and what was contributed to and for the project)?

Also, `CITATION.cff` must be in the root but `CODE_OF_CONDUCT.md` could be moved (my interpretation at least):
> * To make your code of conduct visible in the repository's root directory, type CODE_OF_CONDUCT in the file name field.
> * To make your code of conduct visible in the repository's docs directory, type docs/CODE_OF_CONDUCT.
> * To make your code of conduct visible in the repository's .github directory, type .github/CODE_OF_CONDUCT.

Reply:
I'm working on a branch that moves around the files.

Reply:
See #3462